packages/flutter/lib/src/widgets/scrollable_helpers.dart

```diff
@@ -30,28 +35,118 @@ class ScrollableDetails {
   /// cannot be null.
   const ScrollableDetails({
     required this.direction,
-    required this.controller,
-    this.clipBehavior,
-  });
-
-  /// The direction in which this widget scrolls.
-  ///
-  /// Cannot be null.
+    this.controller,
+    this.physics,
+    @Deprecated(
+      'Migrate to decorationClipBehavior. '
+      'This property was deprecated so that its application is clearer. This clip '
+      'applies to decorators, and does not directly clip a scroll view. '
+      'This feature was deprecated after v3.9.0-1.0.pre.'
+    )
+    Clip? clipBehavior,
+    Clip? decorationClipBehavior,
+  }) : decorationClipBehavior = clipBehavior ?? decorationClipBehavior;
+
+  /// A constructor specific to a [Scrollable] with an [Axis.vertical].
+  const ScrollableDetails.vertical({
+    bool reverse = false,
+    this.controller,
+    this.physics,
+    this.decorationClipBehavior,
+  }) : direction = reverse ? AxisDirection.up : AxisDirection.down;
+
+  /// A constructor specific to a [Scrollable] with an [Axis.horizontal].
+  const ScrollableDetails.horizontal({
+    bool reverse = false,
+    this.controller,
+    this.physics,
+    this.decorationClipBehavior,
+  }) : direction = reverse ? AxisDirection.left : AxisDirection.right;
+
+  /// {@macro flutter.widgets.Scrollable.axisDirection}
   final AxisDirection direction;
 
-  /// A [ScrollController] that can be used to control the position of the
-  /// [Scrollable] widget.
-  ///
-  /// This can be used by [ScrollBehavior] to apply a [Scrollbar] to the associated
-  /// [Scrollable].
-  final ScrollController controller;
+  /// {@macro flutter.widgets.Scrollable.controller}
+  final ScrollController? controller;
```

Comment:
Why is this nullable now? I noticed same new asserts above that assert its not. Maybe add some docs describing in what circumstances this needs to be set?

Reply:
Will do! This is nullable now because it needs to be ok for developers to not provide a scroll controller when using ScrollableDetails to configure a 2D scroller. This allows for things like PrimaryScrollController.

Reply:
Am I right to assume that the 2D scroller will than fill this in with a ScrollController before this reaches the decoration stuff, which assumes this is set?

Reply:
Yup! It will, and that is also why I added the assertions, so any use ensures a controller is set. :)